src/OpFactory.py

```diff
@@ -0,0 +1,50 @@
+import cirq
+import qtree
+
+class OpFactory:
+    pass
+
+
+class CirqFactory:
+    H=cirq.H
+    cX=cirq.CX
+
+    @staticmethod
+    def ZPhase(x, alpha):
+        return cirq.ZPowGate(exponent=float(alpha)).on(x)
+
+    cZ=cirq.CZ
+
+QtreeFactory = qtree.operators
+
+
+class CircuitCreator:
```

Comment:
So the point of these is just to initialize and create a circuit, right? Not to add in additional gates to an already existing one? 
Then I like this paradigm, since it's basically what cirq does, but we get to do it internally much more cleanly

Reply:
Yeah, the `Creator`s create and then composer modifies. I am kinda hesitant on the subdivision of creation of circuits and adding gates to it. Right now Composer inherits the creation methods from Creator.
- Do we actually need this subdivision?
- Should we do this through inheritance? Maybe pass an instance of Creator to the composer on init?

Reply:
Yeah, it's kind of weird to have a division to just create the circuit and then use a different class to add gates to it. I'd argue that given this structure it can make sense. The generic creator to create the circuit, and then we can have different specialized composers. In this case the specialized QAOA composer lets us make the circuit specifically for QAOA. 

I agree with you that there's no need to have that subdivision from the _user_ point of view. An instance of the creator in the composer can allow users to create the whole qaoa circuit using just the QAOA composer class